osdk/src/cli.rs

```diff
@@ -109,6 +115,48 @@ pub struct RunArgs {
     pub cargo_args: CargoArgs,
     #[command(flatten)]
     pub osdk_args: OsdkArgs,
+    #[command(flatten)]
+    pub gdb_server_args: GdbServerArgs,
+}
+
+#[derive(Debug, Args, Clone, Default)]
+pub struct GdbServerArgs {
+    /// Whether to enable QEMU GDB server for debugging
+    #[arg(
+        long,
+        short = 'G',
+        help = "Enable QEMU GDB server for debugging",
+        default_value_t
+    )]
+    pub gdb_server: bool,
+    #[arg(
+        long = "vsc",
+        help = "Generate a '.vscode/launch.json' for debugging with Visual Studio Code \
+                (only works when '--gdb-server' is enabled)",
+        default_value_t
+    )]
+    pub vsc_launch_file: bool,
+    #[arg(
+        long = "gdb-stub",
+        help = "Path of GDB server stub",
+        value_name = "PATH",
+        default_value = "/tmp/aster-gdb-socket"
```

Comment:
The default path should be a relative path to the Rust project root. Consider a OSDK-managed Rust workspace.

```
my-fancy-os-rs/
    Cargo.toml
    OSDK.toml
    my-fancy-kernel/
        Cargo.toml
        OSDK.toml
        src/
             lib.rs
    my-fancy-lib/
        Cargo.toml
        OSDK.toml
        src/
             lib.rs
```

The `cargo osdk run --gdb-server` command could be run under `my-fancy-os-rs/`, `my-fancy-kernel/`, or `my-fancy-lib/`. So in theory, there can be three OSDK-managed, GDB-enabled VMs running three different kernels at the same time. No to mention that there could be other OSDK-managed Rust projects being developed and tested on the same machine. So every file that `OSDK` generates or uses should be per-crate.